daemon/images/image_import.go

```diff
@@ -57,7 +54,19 @@ func (i *ImageService) ImportImage(src string, repository, os string, tag string
 		}
 	}
 
-	config, err := dockerfile.BuildFromConfig(&container.Config{}, changes, os)
+	// Normalize platform - default to the operating system and architecture if not supplied.
+	var p specs.Platform
+	if platform == nil {
+		p = platforms.DefaultSpec()
+	} else {
+		if platform.Architecture == "" {
+			// TODO platforms.Normalize() only sets default OS, but does not set default Architecture if missing.
+			platform.Architecture = runtime.GOARCH
+		}
```

Comment:
Discussing with @tonistiigi to remove this part, and make it the caller's responsibility to fill in the fields